packages/pluggableWidgets/badge-web/src/Badge.tsx

```diff
@@ -1,26 +1,36 @@
-import { ReactNode, useCallback, createElement } from "react";
+import { ReactNode, useCallback, createElement, KeyboardEvent } from "react";
 import { executeAction } from "@widgets-resources/piw-utils";
-import { ValueStatus } from "mendix";
 
 import { BadgeContainerProps } from "../typings/BadgeProps";
 import { Badge as DisplayBadge } from "./components/Badge";
 
-import "./ui/BadgeContainer.css";
-
 export default function Badge(props: BadgeContainerProps): ReactNode {
-    const onClick = useCallback(() => {
-        executeAction(props.onClick);
-    }, [props.onClick]);
+    const onClick = useCallback(() => executeAction(props.onClick), [props.onClick]);
+
+    const onKeyDown = useCallback(
+        (event: KeyboardEvent<HTMLSpanElement>) => {
+            if (event.key === "Enter" || event.key === " ") {
+                onClick();
+            }
+        },
+        [onClick]
+    );
+
+    let actualTabIndex;
+
+    if (props.onClick && props.onClick.canExecute) {
+        actualTabIndex = props.tabIndex ?? 0;
+    }
 
     return (
         <DisplayBadge
             type={props.type}
-            value={props.value && props.value.status === ValueStatus.Available ? props.value.value : ""}
-            brandStyle={props.brandStyle}
-            clickable={props.onClick && props.onClick.canExecute}
-            onClick={onClick}
+            value={props.value?.value ?? ""}
+            onClick={props.onClick && props.onClick.canExecute ? onClick : undefined}
+            onKeyDown={props.onClick && props.onClick.canExecute ? onKeyDown : undefined}
             className={props.class}
             style={props.style}
+            tabIndex={actualTabIndex}
```

Comment:
As you are doing same checks in line 21, 29 and 30, better extract `props.onClick && props.onClick.canExecute` into a const and reuse over the code. Also inline the `tabIndex` logic here `myPreviousCheck ? props.tabIndex ?? 0 : undefined`

Reply:
Done :)